### PR TITLE
[Concurrency] Simplify the type story for `Continuation` by eliminating `Throwing` variants

### PR DESCRIPTION
**Pull request by grynspan**
The goal of doing this is to reduce the amount of boilerplate and repeated code w.r.t. Continuation. Having just added `resume()` in four places, I got the sense that there was a lot of common code that was being duplicated. D.R.Y.

This change resolves rdar://74154769.

### CODE REVIEW

**Comment by grynspan** [comment]
A previous iteration of this code included a `Continuation` protocol that owned, in particular, default implementations of `resume(with:)` and `resume()`. The core team decided against implementing a protocol here.

**Comment by grynspan** [comment]
@swift-ci Please smoke test

**Comment by grynspan** [comment]
Impressively, this is -21 lines of code _despite_ adding lots of markup!

**Comment by grynspan** [comment]
@swift-ci Please smoke test

**Comment by grynspan** [comment]
@swift-ci Please smoke test

**Comment by grynspan** [comment]
@swift-ci Please test

**Comment by grynspan** [comment]
Squashed changes.

@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci Please clean test macOS platform

**Comment by grynspan** [comment]
@swift-ci please test macOS

File: stdlib/public/Concurrency/ContinuationProtocol.swift
Comment (jckarter) [inline stdlib/public/Concurrency/ContinuationProtocol.swift:4]: An underscored protocol is still an ABI liability, and we can't stop outside code from adding further conformers. It'd be better to duplicate the code.

**Review by jckarter — CHANGES_REQUESTED** [review]

File: stdlib/public/Concurrency/ContinuationProtocol.swift
Comment (grynspan) [reply stdlib/public/Concurrency/ContinuationProtocol.swift:4]: Alright. I strongly dislike repeating code but I defer to your expertise. Will change.

**Comment by grynspan** [comment]
@swift-ci please clean test

**Comment by grynspan** [comment]
@swift-ci please clean test

**Review by jckarter — APPROVED** [review]
Looks good, thanks!

**Pull request merged**
